### PR TITLE
Refactoring: Move NSIS files to libs\

### PR DESCRIPTION
**Pull request by ann0see**
**Short description of changes**

Moves the NSIS files to the libs\ folder similar to ASIO-SDK. This is done to clean up the repo and simplify the structure. https://github.com/jamulussoftware/jamulus/pull/2601 was the initial draft, and I plan multiple smaller PRs which are more testable.

CHANGELOG: Refactoring: Moved around files to clean up the repository
**Context: Fixes an issue?**

Related to: https://github.com/jamulussoftware/jamulus/pull/2601

**Does this change need documentation? What needs to be documented and how?**

No. 

**Status of this Pull Request**

Ready for testing and review.

**What is missing until this pull request can be merged?**
Testing. I'm a bit worried since the other PR I mentioned didn't fail, but there were some suspected issues.

## Checklist

-  [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
-  [x] I tested my code and it does what I want
-  [x] My code follows the [style guide](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#source-code-consistency) 
-  [x] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
-  [x] I've filled all the content above

### CODE REVIEW

File: Jamulus.pro [hunk 1]
```diff
@@ -1144,6 +1144,6 @@ contains(CONFIG, "disable_version_check") {
 # be sure to update .github/workflows/coding-style-check.yml and .clang-format-ignore as well.
 CLANG_FORMAT_SOURCES = $$files(*.cpp, true) $$files(*.mm, true) $$files(*.h, true)
 CLANG_FORMAT_SOURCES = $$find(CLANG_FORMAT_SOURCES, ^\(android|ios|mac|linux|src|windows\)/)
-CLANG_FORMAT_SOURCES ~= s!^\(windows/nsProcess/|libs/ASIOSDK2/|src/res/qrc_resources\.cpp\)\S*$!!g
+CLANG_FORMAT_SOURCES ~= s!^\(libs/\(NSIS|ASIOSDK2\)/|src/res/qrc_resources\.cpp\)\S*$!!g
```
Comment (pljones) [inline Jamulus.pro:1147]: Does the `libs` bit need to be complex?  Can't it just get rid of anything under `libs`?

Comment (ann0see) [reply Jamulus.pro:1147]: Yeah. That's valid.

File: Jamulus.pro [hunk 2]
```diff
@@ -1144,6 +1144,6 @@ contains(CONFIG, "disable_version_check") {
 # be sure to update .github/workflows/coding-style-check.yml and .clang-format-ignore as well.
 CLANG_FORMAT_SOURCES = $$files(*.cpp, true) $$files(*.mm, true) $$files(*.h, true)
 CLANG_FORMAT_SOURCES = $$find(CLANG_FORMAT_SOURCES, ^\(android|ios|mac|linux|src|windows\)/)
-CLANG_FORMAT_SOURCES ~= s!^\(windows/nsProcess/|libs/ASIOSDK2/|src/res/qrc_resources\.cpp\)\S*$!!g
+CLANG_FORMAT_SOURCES ~= s!^\(libs/.*/|src/res/qrc_resources\.cpp\)\S*$!!g
```
Comment (pljones) [inline Jamulus.pro:1147]: ... and thinking about it further... does line 1146 filter out or filter in?  If it's filtering out anything that doesn't match that pattern, `libs/` shouldn't match anyway....

Comment (ann0see) [reply Jamulus.pro:1147]: I think it should be a filter out?

@hoffie ?

**Comment by ann0see** [comment]
I just read the following: 

> Building the projects in this solution one at a time. To enable parallel build, please add the "-m" switch.

Probably it's worth adding -m on the nsProcess build.

File: Jamulus.pro (hunk 2 above)
Comment (hoffie) [reply Jamulus.pro:1147]: [`$$find()`](https://doc.qt.io/qt-5/qmake-function-reference.html#find-variablename-substr) keeps all entries which match the pattern (similar to grep). So, yes, seems like having `libs/` in 1147 is redundant (`src/res/...` isn't, though). Should probably be tested though (run `qmake` with the change, `grep clang-format Makefile`).

**Comment by ann0see** [comment]
Ok. I successfully installed JACK and ASIO build on my Win 11 machine.

**Comment by hoffie** [comment]
Does this need updating as well?
```
.github/workflows/autobuild.yml-314-          path: |
.github/workflows/autobuild.yml-315-            C:\Qt
.github/workflows/autobuild.yml-316-            C:\ChocoCache
.github/workflows/autobuild.yml:317:            ${{ github.workspace }}\windows\NSIS
.github/workflows/autobuild.yml-318-            ${{ github.workspace }}\libs\ASIOSDK2
```

We want to cache the downloaded NSIS source, but not (even if we seem to...?) anything coming from our repo or any build artifacts.

**Comment by ann0see** [comment]
.github/workflows/autobuild.yml:317:            ${{ github.workspace }}\windows\NSIS

needs updating. I don't think the other ones have changed?

**Comment by hoffie** [comment]
>  I don't think the other ones have changed?

Sorry, I just added the other lines for context. NSIS is what I meant.

File: Jamulus.pro (hunk 2 above)
Comment (ann0see) [reply Jamulus.pro:1147]:
```
$:~/Proj/jamulus$ make clang_format 
clang-format -i android/androiddebug.cpp src/analyzerconsole.cpp src/audiomixerboard.cpp src/buffer.cpp src/channel.cpp src/chatdlg.cpp src/client.cpp src/clientdlg.cpp src/clientrpc.cpp src/clientsettingsdlg.cpp src/connectdlg.cpp src/levelmeter.cpp src/main.cpp src/multicolorled.cpp src/protocol.cpp src/rpcserver.cpp src/server.cpp src/serverdlg.cpp src/serverlist.cpp src/serverlogging.cpp src/serverrpc.cpp src/settings.cpp src/signalhandler.cpp src/socket.cpp src/util.cpp src/recorder/creaperproject.cpp src/recorder/cwavestream.cpp src/recorder/jamcontroller.cpp src/recorder/jamrecorder.cpp src/sound/soundbase.cpp src/sound/asio/sound.cpp src/sound/coreaudio-mac/sound.cpp src/sound/jack/sound.cpp src/sound/oboe/sound.cpp ios/ios_app_delegate.mm mac/activity.mm mac/badgelabel.mm src/sound/coreaudio-ios/sound.mm ios/ios_app_delegate.h mac/activity.h mac/badgelabel.h src/analyzerconsole.h src/audiomixerboard.h src/buffer.h src/channel.h src/chatdlg.h src/client.h src/clientdlg.h src/clientrpc.h src/clientsettingsdlg.h src/connectdlg.h src/global.h src/levelmeter.h src/multicolorled.h src/protocol.h src/rpcserver.h src/server.h src/serverdlg.h src/serverlist.h src/serverlogging.h src/serverrpc.h src/settings.h src/signalhandler.h src/socket.h src/testbench.h src/threadpool.h src/util.h src/recorder/creaperproject.h src/recorder/cwavestream.h src/recorder/jamcontroller.h src/recorder/jamrecorder.h src/sound/soundbase.h src/sound/asio/sound.h src/sound/coreaudio-ios/sound.h src/sound/coreaudio-mac/sound.h src/sound/jack/sound.h src/sound/oboe/sound.h
```

seems ok?

Comment (hoffie) [reply Jamulus.pro:1147]: Yep.

File: windows/deploy_windows.ps1 [hunk 3]
```diff
@@ -159,7 +159,7 @@ Function Install-Dependencies
     }
     Initialize-Module-Here -m "VSSetup"
     Install-Dependency -Uri $NsisUrl `
-        -Name $NsisName -Destination "NSIS"
+        -Name $NsisName -Destination "..\libs\NSIS\NSIS-source"
```
Comment (pljones) [inline windows/deploy_windows.ps1:162]: Why should this be different from line 328?

Comment (ann0see) [reply windows/deploy_windows.ps1:162]: I tested that and it seemed to just append the value of $RootPath to the working directory

**Review by pljones — APPROVED** [review]

**Pull request merged**